airflow/utils/log/logging_mixin.py

```diff
@@ -193,12 +202,15 @@ def write(self, message):
 
         :param message: message to log
         """
-        if not message.endswith("\n"):
-            self._buffer += message
-        else:
-            self._buffer += message.rstrip()
+        newline = message.endswith("\n")
+        result = message.rstrip() if newline else message
+
+        self._buffer += result
+        if newline:
```

Comment:
Is any reason to change condition here, I guess previous one more readable.

```python
        if not message.endswith("\n"):
            self._buffer += message
        else:
            message = message.rstrip()
            self._buffer += message
            self.flush()
        return len(message)
```

Reply:
Changed it to use an if statement